server/connection.ts

```diff
@@ -0,0 +1,365 @@
+import { Socket } from 'socket.io';
+
+import { Room } from './room';
+import { fetchYoutubeVideo, getYoutubeVideoID } from './utils/youtube';
+import { assignVM, resetVM } from './vm';
+import { PlaylistVideo } from '.';
+
+class Connection {
+  private socket: Socket;
+  private room: Room;
+
+  constructor(socket: Socket, room: Room) {
+    this.socket = socket;
+    this.room = room;
+    this.room.roster.push({ id: socket.id });
+    this.room.io.emit('roster', this.room.roster);
+    this.setupSocketListeners();
+  }
+
+  setupSocketListeners() {
+    this.socket.on('CMD:name', this.setName);
+    this.socket.on('CMD:picture', this.addPicture);
+    this.socket.on('CMD:host', this.host);
+    this.socket.on('CMD:addVideoToPlaylist', this.addVideoToPlaylist);
+    this.socket.on('CMD:play', this.playVideo);
+    this.socket.on('CMD:pause', this.pauseVideo);
+    this.socket.on('CMD:seek', this.seekVideo);
+    this.socket.on('CMD:ts', this.sendVideoTimestamp);
+    this.socket.on('CMD:chat', this.sendChatMessage);
+    this.socket.on('CMD:joinVideo', this.joinVideo);
+    this.socket.on('CMD:leaveVideo', this.leaveVideo);
+    this.socket.on('CMD:joinScreenShare', this.joinScreenShare);
+    this.socket.on('CMD:leaveScreenShare', this.leaveScreenShare);
+    this.socket.on('CMD:startVBrowser', this.startVBrowser);
+    this.socket.on('CMD:stopVBrowser', this.stopVBrowser);
+    this.socket.on('CMD:changeController', this.changeController);
+    this.socket.on('CMD:askHost', this.askHost);
+    this.socket.on('signal', this.emitSignal);
+    this.socket.on('signalSS', this.emitSSSignal);
+    this.socket.on('disconnect', this.disconnect);
+  }
+
+  setName = (data: string) => {
+    if (!data || !this.socket) {
+      return;
+    }
+    if (data && data.length > 100) {
+      return;
+    }
+    this.room.nameMap[this.socket.id] = data;
+    this.room.io.of(this.room.roomId).emit('REC:nameMap', this.room.nameMap);
+  };
+
+  addPicture = (data: string) => {
+    if (!this.socket) {
```

Comment:
It probably doesn't hurt to do so, but is it really necessary to check for this.socket on every method?

Reply:
Not anymore, did this before because we I used a socket which was undefined on init. Will change!